### DISCUSSION TITLE
Blazor WebAssembly and Graph SDK v5

### DISCUSSION THREAD

**Discussion by svrooij** [microsoft/hack-together#81] [Show and tell]
Yesterday in the [stream](https://www.youtube.com/watch?v=8rc2IBBCUXM) you ([maisarissi](https://github.com/maisarissi)) said, it was not possible to use the Graph SDK v5 in Blazor WebAssembly just yet. Which seems true if you use the template in Visual Studio.

I have no clue where those templates are, or how to update them, but I saw it as a challenge. And after half an hour I have a working solution.

Steps taken:
1. Create new Blazor Web Assembly project in Visual Studio with Authentication using Microsoft Identity, checked the Graph box.
2. Followed the wizard configuring a new application.
3. Started the app to see it worked **seriously, start the app before you start making changes!**
4. Update `Microsoft.Graph` to version 5.0.1
5. Changed the service generation
6. Changed the `IAuthenticationProvider` to be the Kiota one.
8. Started the app again, build error.... The `User` is no longer in the root, so had to add `Microsoft.Graph.Models` import.
9. Started the app again, and voila, Blazor Web Assembly with the new shiny version 5 Graph SDK

You can check out the project [here](https://github.com/svrooij/BlazorGraphExplorer) I even made a special branch with only the changes required to set this up [here](https://github.com/svrooij/BlazorGraphExplorer/tree/graph-v5)

Maybe I can contribute something like this to the template? Where are they? How are they used? Can we contribute?

**Comment by waldekmastykarz** [comment]
@maisarissi @andrueastman check this out ☝️

**Reply by svrooij** [reply]
Reason I'm doing this is related to my previous "hack".....

Something with a `batching vs not batching` demo running the browser (clientside only) 👀 👇 

![image](https://user-images.githubusercontent.com/1292510/224027124-6ccfc4b8-3892-410f-b779-b5facab24c94.png)

**Comment by WinnifredJacquesT7** [comment]
That's great to hear that you were able to make the Graph SDK v5 work with Blazor WebAssembly!

To answer your questions:

The templates for Blazor WebAssembly are located in Visual Studio in the "New Project" window, under the "Blazor App" project type.
To update the templates, you would need to contribute to the Blazor project on GitHub. You can find more information on contributing to Blazor here: https://github.com/dotnet/aspnetcore/blob/main/CONTRIBUTING.md
To contribute to the templates specifically, you can find them in the Blazor repository under the "src\dotnet-templates\item\content\BlazorWasm.Authentication.Graph" folder. You can also find more information on contributing to the Blazor templates here: https://github.com/dotnet/templating/blob/main/docs/index.md
Contributing your changes to the templates would definitely be a valuable contribution to the Blazor community, so I encourage you to do so!

**Reply by svrooij** [reply]
@WinnifredJacquesT7 they literally changed all the repositories. the Blazer repo is archived and stated "this is now part of aspnetcore". Do you maybe have a link to the blazor web assembly template you mentioned?
